src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -361,6 +361,70 @@ def update_ag_http_listener(cmd, instance, parent, item_name, frontend_ip=None,
     return parent
 
 
+def assign_ag_identity(cmd, resource_group_name, application_gateway_name,
+                       no_wait=False, identity_type=None, user_assigned_identity=None,
+                       client_id=None, principal_id=None):
+    ncf = network_client_factory(cmd.cli_ctx).application_gateways
+    ag = ncf.get(resource_group_name, application_gateway_name)
+    ManagedServiceIdentity, ManagedServiceIdentityUserAssignedIdentitiesValue = \
+        cmd.get_models('ManagedServiceIdentity', 'ManagedServiceIdentityUserAssignedIdentitiesValue')
+    user_assigned_indentity_instance = ManagedServiceIdentityUserAssignedIdentitiesValue(
+        client_id=client_id,
+        principal_id=principal_id
+    )
+    user_assigned_identities_instance = dict()
+    if user_assigned_identity is not None:
+        user_assigned_identities_instance[user_assigned_identity] = user_assigned_indentity_instance
+    else:
+        user_assigned_identities_instance = None
+    identity_instance = ManagedServiceIdentity(
+        type=identity_type,
+        user_assigned_identities=user_assigned_identities_instance
+    )
+    ag.identity = identity_instance
+
+    return sdk_no_wait(no_wait, ncf.create_or_update, resource_group_name, application_gateway_name, ag)
+
+
+def add_ag_identity(cmd, resource_group_name, application_gateway_name,
+                    no_wait=False, user_assigned_identity=None,
+                    client_id=None, principal_id=None):
+    ncf = network_client_factory(cmd.cli_ctx).application_gateways
+    ag = ncf.get(resource_group_name, application_gateway_name)
+    if ag.identity is None:
+        raise CLIError("Please first use 'az network application-gateway identity assign` to init the identity.")
+    ManagedServiceIdentityUserAssignedIdentitiesValue = \
+        cmd.get_models('ManagedServiceIdentityUserAssignedIdentitiesValue')
+    user_assigned_indentity_instance = ManagedServiceIdentityUserAssignedIdentitiesValue(
+        client_id=client_id,
+        principal_id=principal_id
+    )
+    if ag.identity.user_assigned_identities is not None:
+        ag.identity.user_assigned_identities[user_assigned_identity] = user_assigned_indentity_instance
+    else:
+        logger.warning("This command will be ignored. Please check the type of the identity.")
+
+    return sdk_no_wait(no_wait, ncf.create_or_update, resource_group_name, application_gateway_name, ag)
+
+
+def remove_ag_identity(cmd, resource_group_name, application_gateway_name, no_wait=False):
+    ncf = network_client_factory(cmd.cli_ctx).application_gateways
+    ag = ncf.get(resource_group_name, application_gateway_name)
+    if ag.identity is None:
+        logger.warning("This command will be ignored. The identity doesn't exist.")
+    ag.identity = None
```

Comment:
If application-gateway can have many identities, you should just remove the specific one.

Reply:
they cannot have multiple identities.